### DISCUSSION TITLE
Custom LookupEditor "edit" not working

### DISCUSSION THREAD

**Discussion by axelcasauran** [serenity-is/Serenity#6282] [Help] [answered]
Hi,

I created a custom lookupeditor with InPlaceAdd and I was able to make it work. But the edit functionality is not working, anyone?

![2022-02-20 22_48_51-People - HappyPocket](https://user-images.githubusercontent.com/13443821/154848537-86e4e2fc-58d7-46e9-b4be-e1b771f7bf77.png)

**Comment by wezmag** [comment, marked as answer]
Please share the code of your custom lookupeditor.

**Reply by axelcasauran** [reply]
Hello..

Here is my custom lookupeditor.
![2022-02-21 15_19_27-Window](https://user-images.githubusercontent.com/13443821/154907233-7e1153d0-9416-49f3-bcec-e322b1cd82ff.png)

This is how I call it.
![2022-02-21 15_22_00-Window](https://user-images.githubusercontent.com/13443821/154907560-359180ba-b1f2-4d05-8508-289a4bcbccd4.png)

**Reply by wezmag** [reply]
I think your LookupEditor does not include an Id field, that's why the inPlaceAdd Edit does not work.
The dialog retrieve the data from backend using the Id.
In the lookupeditor constructor, add below: 
``` csharp
IdField = DiscipleshipJourneyRow.Fields.intDiscipleshipJourneyId.PropertyName;
TextField = DiscipleshipJourneyRow.Fields.strDiscipleshipJourney.PropertyName;
```
In the lookupeditor, add below
``` csharp 
protected override void PrepareQuery(SqlQuery query)
{
   var fld = DiscipleshipJourneyRow.Fields;
   query.Select(fld.intDiscipleshipJourneyId, fld.strDiscipleshipJourney)
            .Where(.....) //too lazy to type all your criteria.
            .OrderBy(....) 
}
```
Finally, remove the `GetItems`

Also, I think you can check out this page for lookup script, multi tenancy and cache: https://serenity.is/docs/tutorials/multi_tenancy/handling_lookup_scripts

**Reply by axelcasauran** [reply]
it works like a charm! thank you!
